2020/02/21/consolidate-multiple-sftp-accounts-on-one-master-account.md

```diff
@@ -0,0 +1,154 @@
+---
+author: "Selvakumar Arumugam"
+title: "Consolidate Multiple SFTP Accounts on One Master Account"
+tags: ssh, sftp, shell
+---
+
+
+Recently, a data intensive client implemented a workflow process to generate various reports and insights from a list of facilities. Because a significant portion of these files contain sensitive data, they would need to strictly adhere to HIPAA compliance. Optimally, facilities should be able to transfer files securely and exclusively to our server. Arguably, the best method of achieving this is to create individual SSH File Transfer Protocol (SFTP) accounts for each source.
+
+
+### SFTP Account
+
+Private SFTP accounts were established for each facility and the data was received at a designated path. At these individual points of contact, a third party application picks up the data and processes further into the pipeline. The following demonstrates how SFTP accounts are developed and configured:
+
+
+* Create a user group for SFTP accounts
+
+```bash
+$ addgroup sftpusers
+```
+
+* Configure the following settings in sshd_config (this enables an SFTP account and defaults the location as the home path)
+
+```bash
+$ vi /etc/ssh/sshd_config
+
+Subsystem       sftp    internal-sftp
+
+Match Group sftpusers
+    ChrootDirectory /home/%u
+    AllowTCPForwarding no
+    X11Forwarding no
+    ForceCommand internal-sftp
+```
+
+* Restart SSH Server to apply changes
+
+```bash
+$ /etc/init.d/ssh restart
+```
+
+* Create an SFTP user account for a facility and place in a folder on the home path to receive data
+
+```bash
+useradd $USERNAME
+usermod -g sftpusers -s /usr/sbin/nologin $USERNAME
+mkdir -p /home/$USERNAME
+mkdir -p /home/$USERNAME/INPUT_PATH/
+chown -R root:root /home/$USERNAME
```

Comment:
Since `$USERNAME` is an actual normal environment variable set up by shells, I don't think it's a good variable to use in this example. If you are running as `root`, as I would expect to be doing these things, this will fail because `USERNAME=root`.

Perhaps start this block with:

```
user=the-new-username
```

and then use `$user` throughout?

Reply:
Used lower case.
`sftpuser=the-new-username`